### PR TITLE
feat: add multiple_choice_query parsing

### PR DESCRIPTION
**Pull request by daywiss**
## motivation
we want to add umip 181 mutliple choice query support

## changes
adds a decoder and parser to fetch title, descriptiton and options

image

image

image
image

### CODE REVIEW

File: src/helpers/queryParsing.ts [hunk 1]
```diff
@@ -226,6 +248,62 @@ export function getQueryMetaData(
...
+  title: string;
+  // The full description of the question, optionally using markdown.
+  description: string;
+  // Optionally specify labels and values for each option a user can select.
+  // If not specified the default is ["no", "yes"], which corresponds to prices ['0','1'] in wei.
+  // numbers must be convertible to a signed int256, and specified as integers in base 10.
+  options?: [label: string, value: string][];
+};
+
+const isMultipleChoicQueryFormat = (
```
Comment (daywiss) [inline src/helpers/queryParsing.ts:262]: this was generated by typia but eslint hated it

File: src/helpers/queryParsing.ts [hunk 2]
```diff
@@ -226,6 +248,62 @@ export function getQueryMetaData(
...
+            Array.isArray(elem) &&
+            elem.length === 2 &&
+            "string" === typeof elem[0] &&
+            "string" === typeof elem[1],
+        )));
+  return "object" === typeof input && null !== input && $io0(input);
+};
+
+function decodeMultipleChoiceQuery(decodedAncillaryData: string) {
+  const endOfObjectIndex = decodedAncillaryData.lastIndexOf("}");
```
[the remark this answers is not in the document]
Comment (daywiss) [reply src/helpers/queryParsing.ts:283]: nested objects are fine, but htey wont pass validation.  we expect a certain schema here and we take the last closing bracket which should gaurantee we capture the entire object.  if they dont end with a closing bracket its an invalid schema anyway

File: src/helpers/queryParsing.ts [hunk 3]
```diff
@@ -226,6 +248,62 @@ export function getQueryMetaData(
...
+  const maybeJson =
+    endOfObjectIndex > 0
+      ? decodedAncillaryData.slice(0, endOfObjectIndex)
+      : decodedAncillaryData;
+  const json = JSON.parse(maybeJson);
```
[the remark this answers is not in the document]
Comment (daywiss) [reply src/helpers/queryParsing.ts:288]: yes we catch that and show the error in the ui now
image

File: src/helpers/queryParsing.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gsteenkamp89) [reply src/helpers/queryParsing.ts:300]: oh that makes sense 👍

**Review by gsteenkamp89 — APPROVED** [review]

**Pull request merged**